gammapy/datasets/evaluator.py

```diff
@@ -318,14 +300,10 @@ def _compute_flux_spatial(self):
             if self.geom.region is None or self.psf is None:
                 return 1
 
-            wcs_geom = self.geom.to_wcs_geom(width_min=self.cutout_width).to_image()
+            wcs_geom = self.geom.to_wcs_geom(width_min=self.cutout_width)
+            values = self._compute_flux_spatial_geom(wcs_geom)
 
-            if self.psf and self.model.apply_irf["psf"]:
-                values = self._compute_flux_spatial_geom(wcs_geom)
-            else:
-                values = self.model.spatial_model.integrate_geom(
-                    wcs_geom, oversampling_factor=1
```

Comment:
I understand why you remove here because it is basically unused. But `oversampling=1` is a bug right?

Reply:
yes it is probably a bug, it was not tested before, should I open a separate for that or we backport this PR ?

Reply:
there has been a mix-up when working on https://github.com/gammapy/gammapy/pull/3440

Well I would try to see what happens with the correct oversampling factor.

Reply:
This line is triggered only for region geom when self.model.apply_irf["psf"] is False. If psf is None we directly return 1 (line 301). With this modification and the test added it also returns 1 for the case self.model.apply_irf["psf"] is False.

Reply:
doen in #5390

Reply:
rebased after #5390 so now this PR contains only cleanup